src/_pytest/config/argparsing.py

```diff
@@ -405,6 +405,12 @@ class DropShorterLongHelpFormatter(argparse.HelpFormatter):
     - cache result on action object as this is called at least 2 times
     """
 
+    def __init__(self, *args, **kwargs):
+        """Use more accurate terminal width via pylib."""
+        if "width" not in kwargs:
+            kwargs["width"] = py.io.get_terminal_width()
```

Comment:
should we use `setdefault(...)` and `shutil.get_terminal_width`?

Reply:
`setdefault` maybe, but otherwise it is meant to use an improved version (https://github.com/pytest-dev/py/pull/219).

Reply:
hmmm I'm firmly in the camp that we should stop adding things to pylib

Reply:
sure, but better then to pytest - assuming that Python does not fix it itself.  IIRC there's an old issue / PR idling there.
Anyway, this is merged, and pylib's PR is also idling.  So no worries.. ;)

Reply:
I'd rather put it in pytest, I've been trying to factor out pylib and this does the opposite!

Reply:
ok, maybe later.
Feel free to fix this one up like you suggested initially then if it bothers you.

Reply:
Just FTR, I'm also in the camp that we should avoid adding new things to pylib at all costs, even though dropping it as a dependency is a *far* away dream.